### PR TITLE
[PPML] Add Occlum Attestation to BigDL

### PR DESCRIPTION
**Pull request by piaolaidelangman**
* Update `Dockerfile` to build attestation dependency.
* Update `README.md`.
* Add start server scripts.
* Update `run_spark_on_occlum_glibc.sh` for attestation.

### CODE REVIEW

File: ppml/trusted-big-data-ml/scala/docker-occlum/start-occlum-attestation-server.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	--device=/dev/sgx/enclave \
+	--device=/dev/sgx/provision \
+	-v /var/run/aesmd/aesm.socket:/var/run/aesmd/aesm.socket \
+	-e LOCAL_IP=$LOCAL_IP \
+	-e SGX_MEM_SIZE=8GB \
+	-e PCCS_URL=$PCCS_URL \
+	-e ATTESTATION_SERVER_IP=$ATTESTATION_SERVER_IP \
+	-e ATTESTATION_SERVER_PORT=$ATTESTATION_SERVER_PORT \
+	intelanalytics/bigdl-ppml-trusted-big-data-ml-scala-occlum:2.1.0-SNAPSHOT \
+	bash
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-occlum/start-occlum-attestation-server.sh:18]: bash - > run_attestation_server.sh ?

File: ppml/trusted-big-data-ml/scala/docker-occlum/build_content.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}"  )" >/dev/null 2>&1 && pwd )"
+export DEP_LIBS_DIR="${script_dir}/dep_libs"
+export INITRA_DIR="${script_dir}/init_ra"
+export RATLS_DIR="${script_dir}/../../ra_tls"
+
+function build_ra_tls()
+{
+    rm -rf ${DEP_LIBS_DIR} && mkdir ${DEP_LIBS_DIR}
+    pushd ${RATLS_DIR}
+    bash ./download_and_prepare.sh
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-occlum/build_content.sh:13]: Maybe we can move download _and_prepare into dockerfile

File: ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh [hunk 3]
```diff
@@ -122,7 +129,27 @@ build_spark() {
     mkdir -p image/bin/jars
     cp -f $BIGDL_HOME/jars/* image/bin/jars
     cp -rf /opt/spark-source image/opt/
-    occlum build
+    occlum build --image-key /opt/occlum_spark/data/image_key
+    build_initfs
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh:133]: Need to add ATTESTATION env. Default is false. In that case, we can skip ra_tls

File: ppml/trusted-big-data-ml/scala/docker-occlum/run_attestation_server.sh [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    occlum build
+
+    popd
+}
+
+build_server_instance
+unset HTTPS_PROXY
+if [[ -z "$PCCS_URL" ]]; then
+    echo "NO PCCS"
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-occlum/run_attestation_server.sh:49]: echo "[ERROR] NO PCCS"

Comment (qiyuangong) [reply ppml/trusted-big-data-ml/scala/docker-occlum/run_attestation_server.sh:49]: If PCCS or other env are not set correctly, we should `exit 1 `

File: ppml/trusted-big-data-ml/scala/docker-occlum/build_content.sh (hunk 2 above)
Comment (piaolaidelangman) [reply ppml/trusted-big-data-ml/scala/docker-occlum/build_content.sh:13]: This does execute in the dockerfile. https://github.com/intel-analytics/BigDL/pull/4507/files#diff-edda70a3da505fe823ae7ec2480abf25e94c9cf6bbe5f3baa9bc1caea4f34cfdR229

File: ppml/trusted-big-data-ml/scala/docker-occlum/run_attestation_server.sh (hunk 4 above)
Comment (piaolaidelangman) [reply ppml/trusted-big-data-ml/scala/docker-occlum/run_attestation_server.sh:49]: If add ATTESTATION env, no pccs is not an error.

File: ppml/trusted-big-data-ml/scala/docker-occlum/start-occlum-attestation-server.sh (hunk 1 above)
Comment (piaolaidelangman) [reply ppml/trusted-big-data-ml/scala/docker-occlum/start-occlum-attestation-server.sh]: updated in this https://github.com/intel-analytics/BigDL/pull/4507/files#diff-d7c7a18415f078d25eceb39b1f3e80a4f99694eafdc07cdd4401b2433afe9a24R18

File: ppml/trusted-big-data-ml/scala/docker-occlum/run_attestation_server.sh (hunk 4 above)
Comment (piaolaidelangman) [reply ppml/trusted-big-data-ml/scala/docker-occlum/run_attestation_server.sh:49]: Updated here https://github.com/intel-analytics/BigDL/pull/4507/files#diff-51c5dc18fa529e5ddcf003d1622b47c94d970ac938bed6bf2fd146f0ee039469R86

File: ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh (hunk 3 above)
Comment (piaolaidelangman) [reply ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh]: Updated here https://github.com/intel-analytics/BigDL/pull/4507/files#diff-289171a84c58724029266cd84d480d3a7e4b7718022baa17b9a5866209c2b836R20

Comment (piaolaidelangman) [reply ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh]: I think this is not secure.

File: ppml/trusted-big-data-ml/scala/docker-occlum/build_content.sh (hunk 2 above)
Comment (qiyuangong) [reply ppml/trusted-big-data-ml/scala/docker-occlum/build_content.sh:13]: Nice design

File: ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh [hunk 5]
```diff
@@ -122,7 +131,31 @@ build_spark() {
     mkdir -p image/bin/jars
     cp -f $BIGDL_HOME/jars/* image/bin/jars
     cp -rf /opt/spark-source image/opt/
-    occlum build
+    if [[ $ATTESTATION == "false" ]]; then
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh:134]: if ATTESTATION is not set then ATTESTATION="false"

**Pull request merged**